ros/api/common/utils.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+from flask import request
+from flask_restful import abort
+from ros.lib.models import System, RatingChoicesEnum
+from ros.lib.utils import identity, user_data_from_identity
+
+LOG = logging.getLogger(__name__)
+
+
+def validate_rating_post_api(func):
+    """Validate POST rating request."""
+    allowed_choices = [c.value for c in RatingChoicesEnum]
+
+    def error_msg(error_code, value):
+        errors = {
+            400: "is invalid value for rating.",
+            422: "is invalid choice of input for rating."
+        }
+        return (
+            f"'{value}' { errors.get(error_code, 'Invalid') }"
+            f" Possible values - { *allowed_choices, }"
+        )
+
+    def check_for_rating(data):
+        rating = None
+        try:
+            rating = int(data['rating'])
+        except ValueError:
+            abort(400, message=(error_msg(400, data['rating'])))
+
+        if rating not in allowed_choices:
+            abort(422, message=(error_msg(422, data['rating'])))
+
+        return rating
+
+    def check_for_user():
+        ident = identity(request)['identity']
+        user = user_data_from_identity(ident)
+        username = user['username'] if 'username' in user else None
+
+        if username is None:
+            abort(403, message="Username doesn't exist")
+
+        return username
+
+    def check_for_system(inventory_id):
```

Comment:
An invalid `inventory_id` leads to a 500 error, I have checked the same for `rating` and it works :+1: 

```
api-post-hosts:
	curl  -v -H "Content-Type: application/json"  -H "x-rh-identity: ${b64_identity}" \
 		  -X POST -d '{ "inventory_id": "hello-world", "rating": 1}' \
          "localhost:8005/api/ros/v1/rating" | python -m json.tool
```